src/twitter.ts

```diff
@@ -9,7 +22,27 @@ export const tweetStatus = async (status: string, access_token: string, access_t
         timeout_ms: 60 * 1000,  // optional HTTP request timeout to apply to all requests.
     })
 
-    const tweet = { status }
+    const media_ids = await Promise.all(
+        screenshots.map(async screenshot => {
+            const options = {
+                encoding: 'base64'
+            }
+            const base64Screenshot = await readFile(screenshot.path, options)
+            const { data } = await T.post('media/upload', { media_data: base64Screenshot })
+            return (data as any).media_id_string
+        })
+    )
+
+    const tweet = { status, media_ids }
     const { data, resp } = await T.post('statuses/update', tweet)
-    return { data, resp }
+
+    // Extracting the tweet URL from the data
+    const entities = (data as any).entities
+    const media = entities && entities.media
+    const url = media && media[0] && media[0].url
```

Comment:
We can use typescript to make this easier. Also, we should sanity check the response from twitter. [middleware.trackRequests](https://github.com/morehumaninternet/roar-server/blob/main/src/middleware.ts) will use a status and message of an error thrown from anywhere, so we can simplify this

```typescript
if (resp.statusCode !== 200) {
  throw { status: resp.statusCode, message: resp.statusMessage }
}

const url = (data as any).entities?.media?.[0]?.url
if (!url) {
  throw { message: 'Response from twitter did not include a URL' }
}

return { url }
```